Tests/Runner/DataLoaderTest.php

```diff
@@ -1560,6 +1560,86 @@ public function testWorkspaceCleanupFailure(): void
+                    'input' => 'workspace-snowflake',
+                    'output' => 'workspace-snowflake',
+                ],
+            ],
+        ]);
+        $clientMock = $this->createMock(BranchAwareClient::class);
+        $clientMock->method('verifyToken')->willReturn($this->clientWrapper->getBasicClient()->verifyToken());
+        $workspaceApi = new Workspaces($this->clientWrapper->getBasicClient());
+        $workspaceData = $workspaceApi->createWorkspace(['backend' => 'snowflake']);
+        $workspaces = new Workspaces($this->clientWrapper->getBasicClient());
+        $workspaceGetData = $workspaces->getWorkspace($workspaceData['id']);
+
+        $clientMock->expects(self::once())
+            ->method('apiGet')
+            ->with(sprintf('workspaces/%s', $workspaceData['id']))
+            ->willReturn($workspaceGetData);
+        $clientMock->expects(self::never())
+            ->method('apiDelete');
+
+        $clientWrapperMock = $this->createMock(ClientWrapper::class);
+        $clientWrapperMock->method('getBasicClient')->willReturn($clientMock);
+        $clientWrapperMock->method('getBranchClient')->willReturn($clientMock);
+
+        $logger = new TestLogger();
+        $dataLoader = new DataLoader(
+            $clientWrapperMock,
+            $logger,
+            $this->workingDir->getDataDir(),
+            new JobDefinition(
+                [
+                    'runtime' => [
+                        'backend' => [
+                            'workspace_credentials' => [
+                                'id' => $workspaceData['id'],
+                                'type' => $workspaceData['connection']['backend'],
+                                '#password' => $workspaceData['connection']['password'],
+                            ],
+                        ],
+                    ],
+                ],
+                $component,
+            ),
+            new OutputFilter(10000),
+        );
+
+        $credentials = $dataLoader->getWorkspaceCredentials();
+        self::assertSame(
+            [
+                'host' => $workspaceData['connection']['host'],
+                'warehouse' => $workspaceData['connection']['warehouse'],
+                'database' => $workspaceData['connection']['database'],
+                'schema' => $workspaceData['connection']['schema'],
+                'user' => $workspaceData['connection']['user'],
+                'password' => $workspaceData['connection']['password'],
+                'account' => 'keboola',
+            ],
+            $credentials,
+        );
+        // clean workspace should not do anything (apiDelete is not called)
+        $dataLoader->cleanWorkspace();
+
+        // password is not returned by getWorkspace, otherwise it should be the same
+        unset($workspaceData['connection']['password']);
+        $workspacesData2 = $workspaces->getWorkspace($workspaceData['id']);
+        self::assertSame($workspaceData, $workspacesData2);
```

Comment:
```suggestion
        $workspacesData2 = $workspaces->getWorkspace($workspaceData['id']);
        self::assertSame($workspaceGetData, $workspacesData2);
```

Reply:
nechápu, to nebude fungovat, ne?

Reply:
proc by to nefungovalo? :thinking: Podle toho commentu get vraci stejny data jako create, jen bez hesla. Tak to nemusis porovnavat proti datum insertu (`$workspaceData`), ale proti datum puvodniho getu (`$workspaceGetData`) a neni potreba delat unser, ne?

Reply:
aha, takhle, jo můžu. Já jsem si nevším, že to je $workspaceGetData a ne $workspaceData